### ISSUE TITLE
Difference in filebase64sha256 Function Value Between Terraform 0.11.12 and 0.12.0-beta1

### ISSUE THREAD

**Issue by bflad** [hashicorp/terraform#20652]
### Terraform Version

```console
$ terraform version
Terraform v0.11.12

$ terraform0.12-beta1 version
Terraform v0.12.0-beta1
```

### Console Output

```console
$ terraform console
> base64sha256(file("aws/test-fixtures/lambdatest.zip"))
Ux/n9CP8l+7Ht0tICw0QPs0yLdC1b+1nJ9K5MZR9ENw=
> filebase64sha256("aws/test-fixtures/lambdatest.zip")
Ux/n9CP8l+7Ht0tICw0QPs0yLdC1b+1nJ9K5MZR9ENw=

$ terraform0.12-beta1 console
> filebase64sha256("aws/test-fixtures/lambdatest.zip")
mvtIFEaa1aQob7N+FofXhUwb6FWxfvvxATF9DjEBclW0vsSgxTaNWZQTOGcbPwjFIcfx8BrJzIGVuSrtDhU/xQ==
```

### Expected Behavior

Same file hashing results between Terraform versions.

### Actual Behavior

Different hashes between Terraform 0.11 implementation (correct equivalent hash) and Terraform 0.12 implementation (new incorrect hash).

### Steps to Reproduce

1. `terraform console` from the Terraform AWS Provider repository (or use any binary file for comparison)

### Additional Context

The new `fileXXX()` hashing functions were added due to https://github.com/hashicorp/terraform/issues/20098

It doesn't seem possible to coerce the previous value in Terraform 0.12 when trying `base64encode(filesha256())` and a few other permutations.

### References

- https://github.com/hashicorp/terraform/issues/20098
- https://github.com/hashicorp/terraform/pull/20115
- https://github.com/hashicorp/terraform/pull/20334

**Comment by apparentlymart** [comment]
Hmm... The extra length here seems to suggest this function is using the wrong hash algorithm... sha512 instead of sha256 perhaps. Possibly these functions are just wired into the function table incorrectly.

**Comment by bflad** [comment]
Pull request submitted: #20654

**Comment by bflad** [comment]
I verified the updated code in #20654 returns the previous value:

```console
$ make dev
==> Checking that code complies with gofmt requirements...
GO111MODULE=off go get -u golang.org/x/tools/cmd/stringer
GO111MODULE=off go get -u golang.org/x/tools/cmd/cover
GO111MODULE=off go get -u github.com/golang/mock/mockgen
GOFLAGS=-mod=vendor go generate ./...
2019/03/12 13:04:12 Generated command/internal_plugin_list.go
# go fmt doesn't support -mod=vendor but it still wants to populate the
# module cache with everything in go.mod even though formatting requires
# no dependencies, and so we're disabling modules mode for this right
# now until the "go fmt" behavior is rationalized to either support the
# -mod= argument or _not_ try to install things.
GO111MODULE=off go fmt command/internal_plugin_list.go > /dev/null
go install -mod=vendor .
$ cd ~/src/github.com/terraform-providers/terraform-provider-aws
$ ~/go/bin/terraform console
> filebase64sha256("aws/test-fixtures/lambdatest.zip")
Ux/n9CP8l+7Ht0tICw0QPs0yLdC1b+1nJ9K5MZR9ENw=
```

**Comment by bflad** [comment]
The fix for this has been merged and will release with Terraform 0.12.0-beta2. 👍
